### PR TITLE
fix: alert for crosschain laggy tx

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

this pr fix relayer alert for crosschain laggy tx

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes:
* add each change in a bullet point here
* ...

### CODE REVIEW

**Comment by Azax13** [comment]
?

File: assembler/bsc_assembler.go [hunk 1]
```diff
@@ -138,6 +142,22 @@ func (a *BSCAssembler) process(channelId types.ChannelId) error {
 	}
 	logging.Logger.Debugf("start seq and end enq are %d and %d", startSeq, endSequence)
 
+	a.alertSetMutex.Lock()
```
Comment (guagualvcha) [inline assembler/bsc_assembler.go:145]: Do we really need this `mutex` as I did not see any concurrent call of `process`

File: assembler/bsc_assembler.go [hunk 2]
```diff
@@ -138,6 +142,22 @@ func (a *BSCAssembler) process(channelId types.ChannelId) error {
...
+	if len(a.alertSet) > 0 {
+		var maxTxSeqOfAlert uint64
+		for k := range a.alertSet {
+			if k > maxTxSeqOfAlert {
+				maxTxSeqOfAlert = k
+			}
+		}
+		if startSeq > maxTxSeqOfAlert {
+			a.metricService.SetHasTxDelay(false)
```
Comment (guagualvcha) [inline assembler/bsc_assembler.go:154]: what about removing all the items that is before `startSeq`?

File: assembler/bsc_assembler.go [hunk 3]
```diff
@@ -149,6 +169,12 @@ func (a *BSCAssembler) process(channelId types.ChannelId) error {
 		}
 		status := pkgs[0].Status
 		pkgTime := pkgs[0].TxTime
+		if now-pkgTime > common.TxDelayAlertThreshHold {
```
Comment (guagualvcha) [inline assembler/bsc_assembler.go:172]: `now-pkgTime` --> `time.Since(pkgTime)`

File: assembler/bsc_assembler.go (hunk 2 above)
Comment (alexgao001) [reply assembler/bsc_assembler.go:154]: Actually serving same purpose, as long as there is still alert in the set, keep alerting. Currently Do the celanup once.

File: assembler/bsc_assembler.go (hunk 3 above)
Comment (alexgao001) [reply assembler/bsc_assembler.go:172]: changed

File: assembler/bsc_assembler.go (hunk 1 above)
Comment (alexgao001) [reply assembler/bsc_assembler.go:145]: Yeah its is unnecessary。

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**